cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp

```diff
@@ -1340,6 +1340,14 @@ void Server<ServerConfiguration>::handleAdvertise(const nlohmann::json& payload,
     advertisement.encoding = chan.at("encoding").get<std::string>();
     advertisement.schemaName = chan.at("schemaName").get<std::string>();
 
+    advertisement.schema = chan.find("schema") == chan.end()
+                             ? std::nullopt
+                             : std::optional<std::string>(chan["schema"].get<std::string>());
+    advertisement.schemaEncoding =
+      chan.find("schemaEncoding") == chan.end()
+        ? std::nullopt
+        : std::optional<std::string>(chan["schemaEncoding"].get<std::string>());
+
```

Comment:
I believe the following should also work which avoids a second lookup of the json property

```suggestion
    if (const auto it = chan.find("schema"); it != chan.end()) {
      advertisement.schema = it->get<std::string>();
    }
    if (const auto it = chan.find("schemaEncoding"); it != chan.end()) {
      advertisement.schemaEncoding = it->get<std::string>();
    }
```

Reply:
Some of the previous code accessed fields like this and others used iterators just as you're suggesting, so I just picked one randomly. Same for the another suggestion.

Nonetheless, I agree with both of your points. Should I make a new commit or will you?

Reply:
Ah ok, was not aware of this.